flake.nix

```diff
@@ -17,22 +20,67 @@
     };
   };
 
-  outputs = { self, nixpkgs, flake-utils, cardano-node }:
+  outputs = { self, nixpkgs, flake-utils, cardano-node, poetry2nix }:
     flake-utils.lib.eachDefaultSystem
       (system:
         let
           pkgs = nixpkgs.legacyPackages.${system};
-          python3 = pkgs.python3.override {
-            packageOverrides = self: _: {
-              allure = self.callPackage ./nix/allure.nix { };
-              pytest-allure = self.callPackage ./nix/pytest-allure.nix { };
-              pytest-select = self.callPackage ./nix/pytest-select.nix {};
-              cardano-clusterlib = self.callPackage ./nix/cardano-clusterlib.nix { };
-            };
+          p2n = poetry2nix.legacyPackages.${system};
+          # Workarounds `ModuleNotFoundError: No module named 'poetry'`
+          # cf. https://github.com/nix-community/poetry2nix/blob/master/docs/edgecases.md#modulenotfounderror-no-module-named-packagename
+          pypkgs-build-requirements = {
+            pytest-metadata = [ "poetry" ];
+            pytest-select = [ "poetry" ];
+            m2r2 = [ "setuptools" ];
+            sphinxemoji = [ "setuptools" ];
+            sphinxcontrib-jquery = [ "flit-core" ];
+          };
+          p2n-overrides = p2n.defaultPoetryOverrides.extend (self: super:
+            builtins.mapAttrs
```

Comment:
I find this part really hard to parse. What are we trying to do here?

Reply:
Maybe something like:
```
# For each `package`, and its `build-requirements`, add the `build-requirements` to the `buildInputs`. In the case # where a `build-requirement` is referred to by name, resolve it in the `super` pkg set.
```

However, I still don't understand _why_ this is necessary?

Reply:
Just switched to using sdist instead of building from source (except for pytest): all those overrides for injecting the build tools are not necessary anymore.